src/main/webapp/app/exercises/text/assess/text-submission-assessment.component.ts

```diff
@@ -457,4 +460,16 @@ export class TextSubmissionAssessmentComponent extends TextAssessmentBaseCompone
         super.handleError(error);
         this.saveBusy = this.submitBusy = false;
     }
+
+    /**
+     * Invokes exampleSubmissionService when importExampleSubmission is emitted in assessment-layout
+     */
+    importStudentSubmissionAsExampleSubmission(): void {
+        if (this.submission && this.exercise) {
+            this.exampleSubmissionService.import(this.submission, this.exercise.id!).subscribe(
+                () => this.jhiAlertService.success('artemisApp.exampleSubmission.submitSuccessful'),
+                (error: HttpErrorResponse) => onError(this.jhiAlertService, error),
+            );
+        }
+    }
```

Comment:
it can be refactored later on, but if you can check the buttons in assessment-header.component. The listeners are always defined separately for each exercise type submission component. For this reason, I need to implement same structure for this Import button I introduced newly.

Reply:
I see so there would be major refactoring in the component and templates themselves necessary 👍